### PR TITLE
fix: update CLI callback server to handle redirect flow

### PR DESCRIPTION
**Pull request by aliscott**

### CODE REVIEW

**Review by hugorut — APPROVED** [review]

File: internal/apiclient/auth.go [hunk 1]
```diff
@@ -54,65 +59,52 @@
...
+				w.WriteHeader(400)
+				return
+			}
 
-		if apiKey == "" || state != generatedState {
-			w.WriteHeader(400)
-			apiKey = ""
-		}
-	}))
+			http.Redirect(w, r, redirectTo, http.StatusTemporaryRedirect)
```
[the remark this answers is not in the document]
Comment (vdmgolub) [reply internal/apiclient/auth.go:113]: I guess it is okay as is, but we could also check that `redirectTo` includes `a.Host`.

File: internal/apiclient/auth.go [hunk 2]
```diff
@@ -54,65 +63,68 @@ func (a AuthClient) Login(contextVals map[string]interface{}) (string, string, e
...
 			}
 		}
 	}()
 
-	err := http.Serve(listener, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Set("Access-Control-Allow-Origin", a.Host)
-		w.Header().Set("Access-Control-Allow-Methods", "GET")
-		w.Header().Set("Access-Control-Allow-Headers", "*")
+	go func() {
+		http.Serve(listener, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (vdmgolub) [inline internal/apiclient/auth.go:82]: We could check the error, but maybe we could ignore the errcheck here?

**Review by vdmgolub — APPROVED** [review]

File: internal/apiclient/auth.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aliscott) [reply internal/apiclient/auth.go:113]: A that's probably a better fix than I had, thanks.

File: internal/apiclient/auth.go [hunk 3]
```diff
@@ -54,65 +63,68 @@
...
+				w.WriteHeader(400)
+				return
+			}
 
-		if apiKey == "" || state != generatedState {
-			w.WriteHeader(400)
-			apiKey = ""
-		}
-	}))
+			http.Redirect(w, r, redirectTo, http.StatusTemporaryRedirect)
```
[the remark this answers is not in the document]
Comment (aliscott) [reply internal/apiclient/auth.go:110]: This should be fine now since we validate it against the original login URL

**Pull request merged**